optimum/onnx/graph_transformations.py

```diff
@@ -79,3 +91,217 @@ def remove_duplicate_weights(model: ModelProto, inplace: bool = False) -> ModelP
+            output_shape.append(getattr(dim, "dim_param"))
+        elif getattr(dim, "dim_value"):
+            output_shape.append(getattr(dim, "dim_value"))
+        else:
+            raise ValueError(f"Can not find `dim_param` nor `dim_value` in the output dimension info.")
+
+    return output_shape
+
+
+def _check_num_outputs(model1: ModelProto, model2: ModelProto):
+    if not len(model1.graph.output) == len(model2.graph.output):
+        raise ValueError(
+            f"Two model protos need to have same outputs. But one has {len(model1.graph.output)} "
+            f"outputs while the other has {len(model2.graph.output)} outputs."
+        )
+
+
+def _unify_onnx_outputs(model1: ModelProto, model2: ModelProto):
+    """
+    Unify the outputs of deux ONNX model protos. The outputs of model1 will be replaced by outputs of model2.
+    """
+    _check_num_outputs(model1, model2)
+
+    for idx in range(len(model1.graph.output)):
+        model_output_1 = model1.graph.output[idx]
+        model_output_2 = model2.graph.output[idx]
+        if not model_output_1 == model_output_2:
+            if not (
+                model_output_1.name == model_output_2.name
+                and model_output_1.type.tensor_type.elem_type == model_output_2.type.tensor_type.elem_type
+            ):
+                raise ValueError(
+                    f"Can not match {model_output_1.name} with {model_output_2.name}. Make sure your"
+                    f" model protos have same outputs, have same data types and are in the same order."
+                )
+            model1.graph.output.remove(model_output_1)
+
+            new_output = onnx.helper.make_tensor_value_info(
+                model_output_2.name,
+                model_output_2.type.tensor_type.elem_type,
+                _infer_output_shape(model_output_2),
+            )
+            model1.graph.output.insert(idx, new_output)
+
+    if not all(
+        model_output_1 == model_output_2
+        for model_output_1, model_output_2 in zip(model1.graph.output, model2.graph.output)
+    ):
+        raise RuntimeError(f"Failed to unify outputs of given ONNX model protos.")
+
+
+def _get_all_inputs(model_list: List[ModelProto]):
+    inputs = []
+    input_names = []
+    for model in model_list:
+        for input in model.graph.input:
+            if input.name not in input_names:
+                input_names.append(input.name)
+                inputs.append(input)
+    return inputs
+
+
+def _get_onnx_opset(model: ModelProto):
+    opset_import = model.opset_import[0]
+    return getattr(opset_import, "version")
+
+
+def simplify_onnx_graph(model: ModelProto, print_info: bool = True) -> ModelProto:
+
+    model_simp, check = simplify(model)
+    assert check, "Simplified ONNX model could not be validated."
+    if print_info:
+        print_simplifying_info(model, model_simp)
+
+    return model_simp
+
+
+def _deduplicated_cross_model_initializers(models: List[ModelProto], suffix: str = None):
```

Comment:
```suggestion
def _deduplicated_cross_model_initializers(models: List[ModelProto], suffix: str = ""):
```

Reply:
I have checked if the suffix is None or not in the function `_create_name_sharing_dict`, if we give a "", it will always add a "_" after the original name and it will be ugly.

Reply:
Alright!